server/src/resources/flight_plan/mod.rs

```diff
@@ -31,31 +39,58 @@ impl From<Vec<Row>> for FlightPlans {
             };
             res.push(flight_plan);
         }
-        FlightPlans { flight_plans: res }
+        FlightPlans { list: res }
     }
 }
 
 impl From<Row> for FlightPlanData {
     fn from(fp: Row) -> Self {
-        grpc_debug!("Converting Row to FlightPlanData: {:?}", fp);
+        debug!("Converting Row to FlightPlanData: {:?}", fp);
         let pilot_id: Uuid = fp.get("pilot_id");
         let vehicle_id: Uuid = fp.get("vehicle_id");
         let departure_vertipad_id: Uuid = fp.get("departure_vertipad_id");
         let destination_vertipad_id: Uuid = fp.get("destination_vertipad_id");
         let approved_by: Uuid = fp.get("approved_by");
 
-        //TODO: get vertiport_id based on vertipad_id
-        let departure_vertiport_id: Uuid = Uuid::new_v4();
-        let destination_vertiport_id: Uuid = Uuid::new_v4();
-
-        //TODO: make a function/ macro/ trait to convert from json array to Vec
-        let cargo_weight_g: JsonValue = fp.get("cargo_weight_g");
-        let cargo_weight_g = cargo_weight_g.as_array().unwrap();
-        let cargo_iter = cargo_weight_g.iter();
-        let mut cargo_weight_g: Vec<i64> = vec![];
-        for weight in cargo_iter {
-            cargo_weight_g.push(weight.as_i64().unwrap());
-        }
+        let handle = get_runtime_handle();
+        let vertipad_id = fp.get("departure_vertipad_id");
+        let data = task::block_in_place(move || {
+            handle.block_on(async move {
+                let pool = get_psql_pool();
+                VertipadPsql::new(&pool, vertipad_id).await
+            })
+        });
+        let departure_vertiport_id = match data {
+            Ok(vertipad) => vertipad.id,
+            Err(_) => {
+                panic!(
```

Comment:
It is actually highly unlikely this panic will be called, and if we do end up in this situation, there's a problem with our data integrity.
The conversion function is implemented from a Row type. This Row object is returned directly from the database. The `departure_vertipad_id` and `destination_vertiport_id` are both mandatory fields and have a FK to the `vertipad` table. This means, the vertipads -must- exist and if the select on this uuid doesn't return anything, something weird has happened from the time the flightplan data was returned from the database and the moment this data got enriched by this function.

I'm not sure if we want to gracefully let this go if we get into this situation.

Reply:
I think we want to minimize panics as much as possible for when it's a live system, this would be one broken path in this case where other paths may be operating fine (getting vertiports, pilot info, etc.). I'm not sure we want to shut down the svc-storage microservice due to any single failure of these paths

In a fault protection system you generally have "tiers" of fault protection, so if the schema is unsynced it goes to tier 1 of fault protection (say, logging only essential data to a separate emergency DB). If that fails, it drops to tier 2, etc., so that the rest of the system can continue to do critical work without this path shutting the whole thing down

I'm fine with panicking in initialization, where we learn within the first few seconds if something didn't work

Reply:
I'll approve this for now, but we'll keep this in mind because we may have to implement something like this for redundancy:

```rust
type Execution = (fn() -> Result(), Error>);
enum Operation {
    Nominal(Execution),
    FpTier1(Execution),
    FpTier2(Execution),
    FpTier3(Execution),
}
```

And when critical errors happen we might "elevate" to other execution tiers

Reply:
Approved, when you see this you can resolve and merge in

Reply:
I had already prepared a fix to change the From implementation to TryFrom so we can handle it as an error in case you didn't want to keep the panics.
It's been pushed now